_inc/lib/core-api/wpcom-endpoints/gutenberg-available-extensions.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * Plugin Name: Available Gutenberg Extensions (Blocks and Plugins) for wpcom/v2 WP-API
+ */
+
+class WPCOM_REST_API_V2_Endpoint_Gutenberg_Available_Extensions extends WP_REST_Controller {
+	function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'gutenberg/available-extensions';
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	public function register_routes() {
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( 'Jetpack_Gutenberg', 'get_block_availability' ),
+				'permission_callback' => array( $this, 'get_items_permission_check' ),
+			),
+			'schema' => array( $this, 'get_item_schema' ),
+		 ) );
+	}
+
+	/**
+	 * Return the available Gutenberg extensions schema
+	 *
+	 * @return array Available Gutenberg extensions schema
+	 */
+	public function get_public_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'gutenberg-available-extensions',
+			'type'       => 'object',
+			'properties' => array(
+				'available'          => array(
+					'description' => __( 'Whether the extension is available', 'jetpack' ),
+					'type'        => 'boolean',
+				),
+				'unavailable_reason' => array(
+					'description' => __( 'Human readable reason for the extensiongutenberg/available-extensions not being available', 'jetpack' ),
+					'type'        => 'string',
+				),
+			),
+		);
+
+		return $this->add_additional_fields_schema( $schema );
+	}
+
+	/**
+	 * Ensure the user has proper permissions
+	 *
+	 * @return WP_Error|boolean
+	 */
+	public function get_items_permission_check() {
+		return current_user_can( 'edit_posts' );
```

Comment:
Noting that usually for doing anything with modules (retrieving activation status or (de)activating), we require the `jetpack_manage_modules` capability. Is there any reason why we use a different one here?

Reply:
I think a user can have `edit_posts` but not `jetpack_manage_modules`

Reply:
Okay, but then: if we don't want the users to have access to manage modules, we don't give them the `jetpack_manage_modules` and the `jetpack_admin_page` capabilities, right? But then, with the check being like this, we expose the module information to them, regardless of whether they have the capability. That feels a bit off to me.

Reply:
Your rationale makes sense to me, @tyxla. Changing to `jetpack_manage_modules`.

Reply:
Not so fast @ockham 😆 

FWIW, the modules endpoint needs the `jetpack_admin_page` permission for reading module activation state. It needs `jetpack_manage_modules` for changing the state.

Reply:
I'd defer to someone from @Automattic/jetpack or @Automattic/jetpack-crew to be sure what's best to use here.

Reply:
Can users at "contributor" level manage modules? They [have](https://codex.wordpress.org/Roles_and_Capabilities#Contributor) the `edit_posts` capability.

Reply:
Contrubutors lack the `manage_options` option, which is the one that the Jetpack custom capabilities default to. Therefore, contributors can't manage modules.

Reply:
I agree they should see the blocks. I also don't claim that using `edit_posts` is wrong here.

Note that when Jetpack isn't in dev mode, `jetpack_admin_page` will default to the `read` capability, which is contained by every role (even subscribers).

I only insist that we should check for the same capabilities everywhere where we request module activation state. Otherwise we might be introducing unexpected and difficult to reproduce bugs.

Reply:
> Can users at "contributor" level manage modules? They [have](https://codex.wordpress.org/Roles_and_Capabilities#Contributor) the `edit_posts` capability and thus they should see the blocks.

Right :smile: I think that was my original reasoning.

@tyxla Maybe it makes sense to think of _reading_ module activation status as distinct from _managing_ (changing) it? The point that contributors (or anyone who can `edit_posts`) should be given a correct selection of blocks in the block picker is kinda convincing...

Reply:
I disagree - we use the permissive `read` capability in that case in the Jetpack modules activation state GET endpoint. Why should we use the more restrictive `edit_posts` for reading the same information?

Reply:
My $.02 are that we should be consistent. I think it doesn't make sense to have different permissions to read the same bit of information.

Reply:
...or `jetpack_admin_page`, to literally align with https://github.com/Automattic/jetpack/blob/6ee14cffce0947b7ddfdeb0b58bdda5b7e7edb35/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php#L303

(One last grain of salt: Block availability isn't perfectly congruent with module activation status, since individual blocks can also depend on plans etc.)

Reply:
So if we go by intent (display available blocks in the block picker, i.e. when _editing a post_) rather than implementation (check module activation state -- but also some additional information for individual blocks), `edit_posts` seems to make sense again... :thinking:

Reply:
Note that even though `jetpack_admin_page` falls back to `read`, they're still different capabilities, and if we use `read` instead of `jetpack_admin_page`, we're still introducing subtle differences. One difference could be the fact that in with Jetpack dev mode enabled, `jetpack_admin_page` will default to `manage_options`. Another could be any plugin or custom code that tinkers with the custom capability.

Reply:
Is the TL;DR of the above convo that contributors can still see their blocks? 😅

Reply:
As discussed via Slack, this endpoint won't expose module activation state directly, so using `edit_posts` makes sense 👍